### PR TITLE
Merge *DomainSocket* implementations into other implementations.

### PR DESCRIPTION
**Pull request by normanmaurer**
Motiviation:

We should just merge our *DomainSocket* classes into the "normal" Datagram* and *Socket* classes. By doing so we can eliminate a lot of code and also reduce the class-hierarchy.

Modifications:

- Introduce SocketProtocolFamily and use it
- Merge Channel implementations

Result:

Code cleanup

### CODE REVIEW

File: transport/src/main/java/io/netty5/channel/socket/nio/NioSocketChannel.java [hunk 1]
```diff
@@ -130,6 +131,7 @@ public NioSocketChannel(NioServerSocketChannel parent, EventLoop eventLoop, Sock
                 // Ignore.
             }
         }
+        calculateMaxBytesPerGatheringWrite();
```
Comment (normanmaurer) [inline transport/src/main/java/io/netty5/channel/socket/nio/NioSocketChannel.java:134]: This was a bug... So I re-added the method call

File: transport-native-epoll/src/test/java/io/netty5/channel/epoll/EpollSocketTestPermutation.java [hunk 2]
```diff
@@ -165,13 +168,37 @@ public List<TestsuitePermutation.BootstrapComboFactory<ServerBootstrap, Bootstra
...
+                            @Override
+                            public ServerChannel newChannel(EventLoop eventLoop, EventLoopGroup childEventLoopGroup) {
+                                return new EpollServerSocketChannel(eventLoop, childEventLoopGroup,
+                                        SocketProtocolFamily.UNIX);
+                            }
+
+                            @Override
+                            public String toString() {
+                                return EpollServerSocketChannel.class.getSimpleName()
+                                        + "(..., " + SocketProtocolFamily.UNIX;
```
Comment (chrisvest) [inline transport-native-epoll/src/test/java/io/netty5/channel/epoll/EpollSocketTestPermutation.java:181]: I think you meant to end the strings with closing parens.

Comment (chrisvest) [reply transport-native-epoll/src/test/java/io/netty5/channel/epoll/EpollSocketTestPermutation.java:181]: Similarly in the kqueue test permutations

File: transport-classes-epoll/src/main/java/io/netty5/channel/epoll/EpollDatagramChannel.java [hunk 3]
```diff
@@ -183,6 +202,10 @@ public Future<Void> joinGroup(
         requireNonNull(multicastAddress, "multicastAddress");
         requireNonNull(networkInterface, "networkInterface");
 
+        if (socket.protocolFamily() == SocketProtocolFamily.UNIX) {
```
[the remark this answers is not in the document]
Comment (normanmaurer) [reply transport-classes-epoll/src/main/java/io/netty5/channel/epoll/EpollDatagramChannel.java:205]: not sure I am a big fan of adding such a method. But if you prefer I can do it... Let me know

[the remark this answers is not in the document]
Comment (chrisvest) [reply transport-classes-epoll/src/main/java/io/netty5/channel/epoll/EpollDatagramChannel.java:205]: I'm fine with punting it to see if it becomes a problem or not.

**Comment by normanmaurer** [comment]
Let me merge this so I can make progress... at worse we can do a followup

**Pull request merged**